packages/core/phase2/exceptions/HO200108.ts

```diff
@@ -1,9 +1,50 @@
+import ExceptionCode from "bichard7-next-data-latest/dist/types/ExceptionCode"
+import errorPaths from "../../lib/exceptions/errorPaths"
 import type { AnnotatedHearingOutcome } from "../../types/AnnotatedHearingOutcome"
 import type Exception from "../../types/Exception"
 import type { ExceptionGenerator } from "../../types/ExceptionGenerator"
+import ResultClass from "../../types/ResultClass"
+import checkResultClassExceptions from "./checkResultClassExceptions"
+import areAllPncResults2007 from "../lib/areAllPncResults2007"
+import { areAllResultsOnPnc } from "../lib/generateOperations/areAllResultsOnPnc"
+import checkCaseRequiresRccButHasNoReportableOffences from "../lib/generateOperations/checkCaseRequiresRccButHasNoReportableOffences"
+import hasUnmatchedPncOffences from "../lib/generateOperations/hasUnmatchedPncOffences"
 
-const generator: ExceptionGenerator = (_aho: AnnotatedHearingOutcome, _options): Exception[] => {
-  return []
+const generator: ExceptionGenerator = (aho: AnnotatedHearingOutcome): Exception[] => {
+  const exceptions: Exception[] = []
+  const fixedPenalty = !!aho.AnnotatedHearingOutcome.HearingOutcome.Case.PenaltyNoticeCaseReferenceNumber
+  const allResultsAlreadyOnPnc = areAllResultsOnPnc(aho)
+
+  if (fixedPenalty) {
+    return []
+  }
+
+  checkResultClassExceptions(aho, (offence, result, offenceIndex, resultIndex) => {
+    const ccrId = offence?.CourtCaseReferenceNumber || undefined
+
+    if (!allResultsAlreadyOnPnc && hasUnmatchedPncOffences(aho, ccrId) && !offence.AddedByTheCourt) {
+      return
+    }
+
+    if (result.PNCAdjudicationExists || areAllPncResults2007(aho, ccrId)) {
```

Comment:
`areAllPncResults2007` in this condition is not needed:

```suggestion
    if (result.PNCAdjudicationExists) {
```

Reply:
We can also merge the `if` blocks:

```
    if (result.PNCAdjudicationExists ||
         (!allResultsAlreadyOnPnc && hasUnmatchedPncOffences(aho, ccrId) && !offence.AddedByTheCourt)) {
      return
    }
```